tsdb/db.go

```diff
@@ -621,6 +621,30 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 
 	var wlog *wal.WAL
 	segmentSize := wal.DefaultSegmentSize
+
+	defer func() {
+		// Close files if startup fails somewhere
+		if err != nil {
```

Comment:
```suggestion
		if err == nil {
		  return
		}
```

Reply:
I committed this change but didn't understand what are the benefits. Do you mind explaining :slightly_smiling_face:?

Reply:
losing an indentation level tends to make the code easier to read/understand